datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -1439,6 +1441,57 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_statistics_from_parquet_metadata_dictionary() -> Result<()> {
+        // Data for column c_dic: ["a", "b", "c", "d"]
+        let values = StringArray::from_iter_values(["a", "b", "c", "d"]);
+        let keys = Int32Array::from_iter_values([0, 0, 1, 2]);
+        let dic_array =
+            DictionaryArray::<Int32Type>::try_new(keys, Arc::new(values)).unwrap();
+        let boxed_array: Box<dyn arrow_array::Array> = Box::new(dic_array);
+        let c_dic: ArrayRef = Arc::from(boxed_array);
+
+        // Define the schema
+        let field = Field::new(
+            "c_dic",
+            DataType::Dictionary(Box::new(DataType::Int32), Box::new(DataType::Utf8)),
+            false,
+        );
+        let schema = Schema::new(vec![field]);
+        // Create the RecordBatch
+        let batch1 = RecordBatch::try_new(Arc::new(schema), vec![c_dic]).unwrap();
+
+        // Use store_parquet to write each batch to its own file
+        // . batch1 written into first file and includes:
+        //    - column c_dic that has 4 rows with no null. Stats min and max of string column is missing for this test even the column has values
+        let store = Arc::new(LocalFileSystem::new()) as _;
```

Comment:
what is this needed for?

Reply:
This was from my original test example: https://github.com/apache/datafusion/issues/11145#issue-2378475063

I believe `store` is used to mimic an object store so that we can get table schema from a parquet file